resonance/cross/windowize_by_events.py

```diff
@@ -0,0 +1,151 @@
+import resonance.si as si
+import resonance.db as db
+from resonance.internal import declare_transformation, Processor
+import numpy as np
+import numpy_ringbuffer as np_rb
+import math
+
+@declare_transformation
+# Split data stream to windows using event stream
+# @param data Data stream
+# @param events Events stream
+# @param windowSize Size of resulting window
+# @param dropLateEvents Don't expand buffer infinitely, lateTime controls buffer size and any events that arrive
+#                       with timestamp earlier than last data timestamp-lateTime potentially can be dropped
+# @param lateTime - allowed delay for events (in seconds)
+# @return window
+class windowize_by_events(Processor):
+
+    def __init__(self):
+        self._signal = None
+        self._pointer = None
+        self._times = None
+        self._lastTS = None
+        self._lastSample = None
+        self._events = None
+        self._windowSelector = None
+        self._samplingRate = None
+        self._si = None
+        self._maxBufferSize = None
+        self._shiftBufferTo = None
+        self._window_size = None
+        self._shift = None
+
+    def prepare(self, input_stream, events, window_size, shift=0, drop_late_events=True, late_time=10):
```

Comment:
I am not sure what will happen with drop_late_events=False, we need at least unit test for it